src/containers/products/item-content.svelte

```diff
@@ -20,16 +20,19 @@
   let animation = false;
 
   function purchase() {
-    if (!productControl.productSized || selectedVarietyID != null) {
-      err = false;
-      dispatch('purchase', { varietyID: selectedVarietyID });
-    } else {
+    // not in sizes, select variety automatically
+    if (!productControl.productSized) {
+      const selectedVariety = productControl.varietiesByColor.get(selectedColor);
```

Comment:
This could be confusing, since from the point of view of code, a variety is a pair (color, size), while from the point of view of the user a variety is categorized only by color, and a variety has different sizes. I suggest naming this variable to reflect the sense that this is actually a selected color which acts as a grouping parameter for varieties

Reply:
What do you suggest? I didn't quite understand that map object

Reply:
I suggest renaming the `selectedVariety` variable to anything that would reflect that this is in fact an array of varieties